### PR TITLE
JBPM-8257:Configurable Color coding for executed flows in Process model

### PR DESCRIPTION
**Pull request by nmirasch**
added new process instance diagram color customization parameters

### CODE REVIEW

**Comment by nmirasch** [comment]
@cristianonicolai @mswiderski @jstastny-cz @MarianMacik could you take a look? Thanks!

File: jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/SVGProcessor.java [hunk 1]
```diff
@@ -11,6 +11,11 @@
 
     void transform(Transformation t);
 
+    void defaultCompletedTransformation(String nodeId, String completedNodeColor,
```
Comment (cristianonicolai) [inline jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/SVGProcessor.java]: no need to break line?

File: jbpm-process-svg/src/main/java/org/jbpm/process/svg/SVGImageProcessor.java [hunk 2]
```diff
@@ -58,19 +62,25 @@ public SVGProcessor getProcessor() {
     //Static methods to keep backward compatibility
 
     public static String transform(InputStream svg, List<String> completed, List<String> active) {
-        return transform(svg, completed, active, null);
+        return transform(svg, completed, active, null, COMPLETED_COLOR, COMPLETED_BORDER_COLOR, ACTIVE_COLOR);
+    }
+
+    public static String transform(InputStream svg, List<String> completed, List<String> active, Map<String, String> subProcessLinks){
+        return transform(svg, completed,active,subProcessLinks, COMPLETED_COLOR, COMPLETED_BORDER_COLOR, ACTIVE_COLOR);
```
Comment (cristianonicolai) [inline jbpm-process-svg/src/main/java/org/jbpm/process/svg/SVGImageProcessor.java]: missing space before _subProcessLinks_

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/UIServicesClientImpl.java [hunk 3]
```diff
@@ -210,6 +210,44 @@ public String getProcessInstanceImage(String containerId, Long processInstanceId
...
+            headers.put("Accept", MediaType.APPLICATION_SVG_XML);
+
+            return makeHttpGetRequestAndCreateRawResponse(
+                    build(loadBalancer.getUrl(), IMAGE_URI + "/" + PROCESS_INST_IMG_GET_URI + "?" + params.toString(),
+                          valuesMap), headers);
+        } else {
+            CommandScript script = new CommandScript(Collections.singletonList(
+                    (KieServerCommand) new DescriptorCommand("ImageService",
+                                                             "getActiveProcessImage",
+                                                             new Object[]{containerId,
```
Comment (cristianonicolai) [inline kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/UIServicesClientImpl.java]:
```suggestion
                                                             new Object[]{
                                                                 containerId,
```

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/main/java/org/kie/server/remote/rest/jbpm/ui/ImageResource.java [hunk 4]
```diff
@@ -93,13 +99,18 @@ public Response getProcessImage(@javax.ws.rs.core.Context HttpHeaders headers,
...
+                                            @ApiParam(value = "svg completed node color", required = false, example = COMPLETED_COLOR) @QueryParam(SVG_NODE_COMPLETED_COLOR) @DefaultValue(COMPLETED_COLOR) String svgNodeCompletedColor,
+                                            @ApiParam(value = "svg completed node border color", required = false, example = COMPLETED_BORDER_COLOR) @QueryParam(SVG_NODE_COMPLETED_BORDER_COLOR) @DefaultValue(COMPLETED_BORDER_COLOR) String svgNodeCompletedBorderColor,
+                                            @ApiParam(value = "svg active node color", required = false, example = ACTIVE_COLOR) @QueryParam(SVG_NODE_ACTIVE_COLOR) @DefaultValue(ACTIVE_COLOR) String svgNodeActiveColor) {
         Variant v = getVariant(headers);
         Header conversationIdHeader = buildConversationIdHeader(containerId, context, headers);
         try {
-
-            String svgString = imageServiceBase.getActiveProcessImage(containerId, procInstId);
+             String svgString = imageServiceBase.getActiveProcessImage(containerId, procInstId,
```
Comment (cristianonicolai) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/main/java/org/kie/server/remote/rest/jbpm/ui/ImageResource.java]: just check the formatting here that is a bit off.

File: jbpm-process-svg/src/main/java/org/jbpm/process/svg/SVGImageProcessor.java (hunk 2 above)
Comment (nmirasch) [reply jbpm-process-svg/src/main/java/org/jbpm/process/svg/SVGImageProcessor.java]: ok

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/UIServicesClientImpl.java (hunk 3 above)
Comment (nmirasch) [reply kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/UIServicesClientImpl.java]: @cristianonicolai not sure if I applied well the suggestion, I just put the first value at the next line but it doesn't fit just under Object word like I understand you are proposing. I can do it manually but I looks bad fight against the IDE code formatting.

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/main/java/org/kie/server/remote/rest/jbpm/ui/ImageResource.java (hunk 4 above)
Comment (nmirasch) [reply kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/main/java/org/kie/server/remote/rest/jbpm/ui/ImageResource.java]: yes, fixed

File: jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/JBPMDesignerSVGProcessor.java [hunk 5]
```diff
@@ -18,13 +18,13 @@ public JBPMDesignerSVGProcessor(Document svgDocument, boolean mapById) {
     }
 
     @Override
-    public void defaultCompletedTransformation(String nodeId) {
-        transform(new SetBackgroundColorTransformation(nodeId, COMPLETED_COLOR));
+    public void defaultCompletedTransformation(String nodeId,String completedNodeColor, String completeBorderColor) {
+        transform(new SetBackgroundColorTransformation(nodeId, completedNodeColor));
```
[the remark this answers is not in the document]
Comment (nmirasch) [reply jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/JBPMDesignerSVGProcessor.java]: @jstastny-cz  I thought the same, I just wanted to keep the previous behaviour that seems din't apply that completed border color parameter. For this, I just have added :
transform(new SetBorderColorTransformation(nodeId, completedBorderColor));
The transformation to set that completed node border color on that JBPMDesigner diagrams.

File: jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/StunnerSVGProcessor.java [hunk 6]
```diff
@@ -24,23 +24,23 @@ public StunnerSVGProcessor(Document svgDocument) {
...
     }
 
     @Override
-    public void defaultActiveTransformation(String nodeId) {
+    public void defaultActiveTransformation(String nodeId, String activeNodeColor) {
         transform((summary) -> {
             Optional.ofNullable(summary.getNode(nodeId)).ifPresent(node -> {
                 Optional.ofNullable(node.getBorder()).ifPresent(border -> {
-                    setNodeBorderColor(node.getRenderType(), border, ACTIVE_COLOR);
+                    setNodeBorderColor(node.getRenderType(), border, activeNodeColor);
```
[the remark this answers is not in the document]
Comment (nmirasch) [reply jbpm-process-svg/src/main/java/org/jbpm/process/svg/processor/StunnerSVGProcessor.java]: @jstastny-cz I renamed the variables related to explain that paramenter affects  to the active node border color. I think now is more understandable than activeColor. Could you take a look?

**Comment by nmirasch** [comment]
@cristianonicolai @jstastny-cz changes done, could you take a look? Thanks!

**Comment by cristianonicolai** [comment]
@mswiderski can you take a look, thanks

**Pull request merged**